### PR TITLE
docs: update server, vars, mock docs

### PR DESCRIPTION
**Pull request by yimingjfe**
### <samp>🤖 Generated by Copilot at 170e76c</samp>

The pull request improves the documentation of Modern.js, especially the guides on web server, environment variables, and mock features. It fixes typos, grammatical errors, and formatting issues, and renames and relocates some files for better clarity and consistency. It also updates some section titles and content to reflect the latest features and usage of Modern.js.

### <samp>🤖 Generated by Copilot at 170e76c</samp>

*  Move and update the files on custom web server and mock features from basic-features to advanced-features directory, and improve the grammar and clarity of the content ([link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L7-R15),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L23-R25),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L50-R50),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L64-R68),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L97-R97),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-93f87dd6fa53cac7fe1120ed891d6da327000c5356b7fa0535e3f692e4a770d3L106-R108),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L13-R13),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L19-R21),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L36-R36),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L68-R68),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L78-R80),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L88-R88),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L100-R100),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L140-R140),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L152-R152),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L161-R161),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L182-R182),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL7-R11),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL25-R29),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL35-R35),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL43-R47),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL59-R59),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL66-R68),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-0cf74b8280ff8305eff5bac7b8adca79ce4d968cfde6b2ac1c90de0eaf92b70dL95-R95),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-6a856d0791584a829eb368665b4d855b595a75629588dd31986686a2c563884cL9-R9),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-6a856d0791584a829eb368665b4d855b595a75629588dd31986686a2c563884cL66-R66))
* Modify the description of the `ASSET_PREFIX` and `NODE_ENV` environment variables in `env-vars.mdx`, and improve the grammar and clarity of the content ([link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L13-R13),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L19-R21))
* Modify the description of the `MODERN_TARGET` environment variable and the dead code elimination process in `env-vars.mdx`, and improve the grammar and clarity of the content ([link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L36-R36),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L68-R68))
* Modify the titles of the sections on specifying custom environment variables via shell and via .env file in `env-vars.mdx`, and improve the grammar and clarity of the content ([link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L78-R80),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L88-R88),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L100-R100))
* Modify the description of the `source.globalVars` and `source.define` configurations in `env-vars.mdx`, and improve the grammar and clarity of the content ([link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L140-R140),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L152-R152),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L161-R161),[link](https://github.com/web-infra-dev/modern.js/pull/3905/files?diff=unified&w=0#diff-1c2ce96d2857324efb038f851f66f5c3bfaa6000cc5dc3f3701cf365f666dce1L182-R182))

## Checklist

- [ ] I have added changeset via `pnpm run change`.
- [ ] I have updated the documentation.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

File: packages/document/main-doc/docs/en/guides/basic-features/mock.mdx [hunk 1]
```diff
@@ -4,11 +4,11 @@ sidebar_position: 6
...
 # Mock
 
-Modern.js provides the ability to quickly generate Mock data, allowing the front-end to develop independently without being blocked by the back-end interface.
+Modern.js allows you to easily generate mock data so that the front-end can develop independently without depending on the back-end API.
 
-## Mock File
+## Enabling Mock
 
-By convention, when there is `index.[jt]s` in the `config/mock/` directory, the Mock Data will be automatically enabled, as follows:
+By convention, when there is an index.[jt]s in the config/mock/ directory, Mock data will be automatically enabled:
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/basic-features/mock.mdx:11]:
```suggestion
By convention, when there is an `index.[jt]s` in the `config/mock/`directory,  mock data will be automatically enabled:
```

File: packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx [hunk 2]
```diff
@@ -10,15 +10,15 @@ Modern.js provides support for environment variables, including built-in environ
...
 ### NODE_ENV
 
 The current execution environment and is a **read-only** environment variable whose have different values under different execution commands:
 
-- `production`: the default value when exec `modern build` or `modern serve`.
-- `test`: the default value when exec `modern test`.
-- `development`: the default value when exec `modern dev`, alse the default value of other case.
+- `production`: Default value when running `modern build` or `modern serve`.
+- `test`: Default value when running `modern test`.
+- `development`: Default value when running modern dev, also the default value in other cases.
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx:21]:
```suggestion
- `development`: Default value when running `modern dev`, also the default value in other cases.
```

File: packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx [hunk 3]
```diff
@@ -75,17 +75,17 @@ If you do not use `process.env.MODERN_TARGET` as described above, the code minim
 
 ## Custom Environment Variables
 
-Custom environment variables can be specified in both `shell` and `.env` files.
+You can specify custom environment variables in both shell and .env files.
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx:78]:
```suggestion
You can specify custom environment variables in both `shell` and `.env` files.
```

File: packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx [hunk 4]
```diff
@@ -97,7 +97,7 @@ BAR=456
 The `.env` file follows the following loading rules:
 
 - `.env`: default.
-- `.env.{ MODERN_ENV | NODE_ENV }`: Setting environment variables for a specific environment overrides the same in `.env`.
+- `.env.{ MODERN_ENV | NODE_ENV }`: Overrides .env for a specific environment.
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx:100]:
```suggestion
- `.env.{ MODERN_ENV | NODE_ENV }`: Overrides `.env` for a specific environment.
```

File: packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx [hunk 5]
```diff
@@ -137,7 +137,7 @@ In custom HTML templates, you can also use such environment variables directly.
 
 ### Any Other Names
 
-If you need to use environment variables with any other names in your code, you can config [`source.globalVars`](/configure/app/source/global-vars), for example:
+If you need to use environment variables with any other names in your code, you can configure [`source.globalVars`](/configure/app/source/global-vars), for example:
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/basic-features/env-vars.mdx:140]:
```suggestion
If you need to use environment variables with any other names in your code, you can configure them in [`source.globalVars`](/configure/app/source/global-vars). For example:
```

File: packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx [hunk 6]
```diff
@@ -4,25 +4,25 @@ sidebar_position: 3
 ---
 # Custom Web Server
 
-As a client side-centric development framework, Modern.js has weak customization capabilities on the server side. In some development scenarios, special server level logic needs to be customized, such as user authentication, request preprocessing, and adding page rendering skeletons.
+As a client-centric development framework, Modern.js has weak server-side customization capabilities. In some development scenarios, special server-level logic needs to be customized, such as user authentication, request preprocessing, and adding page rendering skeletons.
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx:7]:
```suggestion
As a client-centric development framework, Modern.js has limited customization capabilities on the server side. However, in some development scenarios, special server-level logic needs to be customized, such as user authentication, request preprocessing, and adding page rendering skeletons.
```

File: packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx [hunk 7]
```diff
@@ -4,25 +4,25 @@ sidebar_position: 3
...
+The first way is to customize the server-side at a specific lifecycle through the server-side runtime API provided by Modern.js. The purpose of providing this way is that in some cases, developers do not need to control the full Web Server, but only need to add server-level logic.
 
-Because the full web server cannot be controlled this way, and the extension logic **only takes effect when the page is requested**. Therefore, it is relatively simple to apply to server level logic, and you do not want to create additional BFFs or BFFs and pages without common server level logic scenarios.
+Because the full web server cannot be controlled this way, and the extension logic **only takes effect when the page is requested**. Therefore, it is relatively simple to apply server-level logic, and you do not want to create additional BFFs or BFFs and pages without common server-level logic scenarios.
 
 You can execute the'pnpm run new 'command in the project root directory to enable the "Custom Web Serve" function:
```
Comment (caohuilin) [inline packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx:27]:
```suggestion
You can run the'pnpm run new 'command in the project root directory to enable the "Custom Web Serve" function:
```

File: packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx [hunk 8]
```diff
@@ -94,7 +94,7 @@ For more detail, see [Middleware] (/apis/app/runtime/web-server/middleware).
 Projects should have the following best practices when using Middleware:
 
 1. In Middleware, you can directly operate origin request and response objects, do event tracking, and inject Node services (databases, Redis, etc.) that may be used for SSR rendering.
-2. Marking and crawler optimization can be done in Middleware.
+2. Functions such as marking and crawler optimization can be done in Middleware.
```
Comment (zllkjc) [inline packages/document/main-doc/docs/en/guides/advanced-features/web-server.mdx:97]: "operations" be more appropriate than "Functions" ?

**Review by caohuilin — APPROVED** [review]

**Pull request merged**